src/core/state/hiddenAssets/hiddenAssets.ts

```diff
@@ -0,0 +1,52 @@
+import create from 'zustand';
+
+import { ParsedUserAsset } from '~/core/types/assets';
+import { SearchAsset } from '~/core/types/search';
+
+import { createStore } from '../internal/createStore';
+
+type UpdateHiddenAssetArgs = {
+  uniqueId: string;
+};
+
+type UpdateHiddenAssetFn = ({ uniqueId }: UpdateHiddenAssetArgs) => void;
+
+export interface HiddenAssetState {
+  hiddenAssets: string[];
+  addHiddenAsset: UpdateHiddenAssetFn;
+  removeHiddenAsset: UpdateHiddenAssetFn;
+}
+
+export const computeUniqueIdForHiddenAsset = (
+  asset: ParsedUserAsset | SearchAsset,
+) => {
+  return `${asset.address}-${asset.chainId}`;
+};
```

Comment:
Is there a particular reason you're building this yourself? Now that https://github.com/rainbow-me/browser-extension/commit/b1e3f5edc7d328978f07338eab49025b372c5f28 is merged, would it make more sense to grab this from the asset itself? Haven't fully thought this through, but would be curious to hear your reasoning.